env0/resource_environment_discovery_configuration.go

```diff
@@ -197,62 +196,31 @@ func discoveryValidatePutPayload(putPayload *client.EnvironmentDiscoveryPutPaylo
 	terragruntVersionSet := putPayload.TerragruntVersion != ""
 
 	switch putPayload.Type {
-	case "opentofu":
+	case client.OPENTOFU:
 		if !opentofuVersionSet {
 			return errors.New("'opentofu_version' not set")
 		}
-	case "terraform":
+	case client.TERRAFORM:
 		if !terraformVersionSet {
 			return errors.New("'terraform_version' not set")
 		}
-	case "terragrunt":
+	case client.TERRAGRUNT:
 		if !terragruntVersionSet {
 			return errors.New("'terragrunt_version' not set")
 		}
 
-		if putPayload.TerragruntTfBinary == "opentofu" && !opentofuVersionSet {
+		if putPayload.TerragruntTfBinary == client.OPENTOFU && !opentofuVersionSet {
 			return errors.New("'terragrunt_tf_binary' is set to 'opentofu', but 'opentofu_version' not set")
 		}
 
-		if putPayload.TerragruntTfBinary == "terraform" && !terraformVersionSet {
+		if putPayload.TerragruntTfBinary == client.TERRAFORM && !terraformVersionSet {
 			return errors.New("'terragrunt_tf_binary' is set to 'terraform', but 'terraform_version' not set")
 		}
-	case "workflow":
+	case client.WORKFLOW:
 	default:
 		return fmt.Errorf("unhandled type %s", putPayload.Type)
 	}
 
-	vcsCounter := 0
```

Comment:
removed this because this may cause some issues. Not clear how to resolve.
Schema validation is a nice to have. Instead BE will error on this.